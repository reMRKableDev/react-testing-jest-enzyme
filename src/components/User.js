import React from "react";
import "./styles/User.css";

function User({ user }) {
  return (
    <article className="user_container">
      <div className="user">
        <h2>User # {user.id}</h2>
        <p>
          <strong>Name:</strong> {user.name}
        </p>
        <p>
          <strong>Email:</strong> {user.email}
        </p>
      </div>
    </article>
  );
}

export default User;
