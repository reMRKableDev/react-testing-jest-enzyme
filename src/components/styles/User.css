.user_container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row wrap;
  margin: 15px 0;
}

.user {
  border: 2px solid #000;
  width: 50%;
  box-shadow: 0.5em 0.5em 0 0 #282c34, inset 0.5em 0.5em 0 0 #61dafb;
}
