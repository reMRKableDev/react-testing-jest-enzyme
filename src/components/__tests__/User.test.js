import React from "react";
import User from "../User";
import { shallow } from "enzyme";

test("shallow renders component with object", () => {
  const fakeUser = {
    id: 255,
    name: "Malcolm Kente",
    email: "malcolm.kente@remrkabledev.com"
  };
  
  const wrapper = shallow(<User user={fakeUser} />);
  expect(wrapper).toMatchSnapshot();
});
