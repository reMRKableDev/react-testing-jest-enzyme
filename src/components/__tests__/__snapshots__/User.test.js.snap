// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders component with object 1`] = `
<article
  className="user_container"
>
  <div
    className="user"
  >
    <h2>
      User # 
      255
    </h2>
    <p>
      <strong>
        Name:
      </strong>
       
      Malcolm Kente
    </p>
    <p>
      <strong>
        Email:
      </strong>
       
      malcolm.kente@remrkabledev.com
    </p>
  </div>
</article>
`;
